### PR TITLE
Fix template websocket

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
For templates, TeamMembers need to be handled similar to ChannelMembers for an associated channel on a board. This PR retrieves the TeamMembers and updates the memberships via websocket for Templates when the board type (Private/Open) changes.

#### Ticket Link

  Fixes https://github.com/mattermost/focalboard/issues/3645

### CODE REVIEW

File: mattermost-plugin/server/manifest.go [hunk 1]
```diff
@@ -45,7 +45,8 @@ const manifestStr = `
         "type": "bool",
         "help_text": "This allows board editors to share boards that can be accessed by anyone with the link.",
         "placeholder": "",
-        "default": false
+        "default": false,
+        "hosting": ""
```
Comment (sbishel) [inline mattermost-plugin/server/manifest.go:49]:
```suggestion
        "default": false
```

File: server/app/boards.go [hunk 2]
```diff
@@ -541,8 +602,8 @@ func (a *App) DeleteBoardMember(boardID, userID string) error {
 	}
 
 	a.blockChangeNotifier.Enqueue(func() error {
-		if synteticMember, _ := a.store.GetMemberForBoard(boardID, userID); synteticMember != nil {
-			a.wsAdapter.BroadcastMemberChange(board.TeamID, boardID, synteticMember)
+		if syntheticMember, _ := a.GetMemberForBoard(boardID, userID); syntheticMember != nil {
```
Comment (sbishel) [inline server/app/boards.go:605]: This fixes a bug a user gets removed when membership is deleted...even when they retain a synthetic membership.

**Comment by sbishel** [comment]
@mgdelacroix - Gentle ping for review. Thanks

File: server/app/boards.go [hunk 3]
```diff
@@ -338,46 +336,83 @@ func (a *App) PatchBoard(patch *model.BoardPatch, boardID, userID string) (*mode
...
+				members, err := a.GetMembersForBoard(updatedBoard.ID)
+				if err != nil {
+					a.logger.Error("Unable to get the board members", mlog.Err(err))
+				}
+				for _, member := range members {
+					if member.Synthetic {
+						a.wsAdapter.BroadcastMemberChange(updatedBoard.TeamID, member.BoardID, member)
+					}
+				}
+			} else if *patch.ChannelID == "" {
```
Comment (mgdelacroix) [inline server/app/boards.go:360]: nit for simplicity
```suggestion
			} else {
```

File: server/app/boards.go [hunk 4]
```diff
@@ -338,46 +336,83 @@ func (a *App) PatchBoard(patch *model.BoardPatch, boardID, userID string) (*mode
...
 }
 
+func (a *App) postChannelMessage(message, channelID string) {
+	err := a.store.PostMessage(message, "", channelID)
+	if err != nil {
+		a.logger.Error("Unable to post the link message to channel", mlog.Err(err))
+	}
+}
+
+func (a *App) broadcastTeamUsers(teamID, boardID string, boardType model.BoardType, members []*model.BoardMember) {
```
Comment (mgdelacroix) [inline server/app/boards.go:393]: It took me a bit to understand the usage of this method, so I'd suggest to add a docline stating the preconditions, something like:

```suggestion
// broadcastTeamUsers notifies the members of a team when a template changes its type
// from public to private or viceversa
func (a *App) broadcastTeamUsers(teamID, boardID string, boardType model.BoardType, members []*model.BoardMember) {
```

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1: non blocking, I think debug lines were left behind on https://github.com/mattermost/focalboard/pull/3747/commits/91295efbf26c9c4758bf68904fc552341f5217ba

**Pull request merged**